### DISCUSSION TITLE
Storing the entire Move struct?

### DISCUSSION THREAD

**Discussion by jongdetim** [SebLague/Chess-Challenge#436] [General]
I was wondering if those of you that are using a transposition table store the entire Move struct as the best move in a position, as it takes up quite a bit (40+ bytes) of memory, or if you for instance only store the move starting and ending square? I think when using only those (2 bytes total), it should be possible to compare them versus the possible moves in our current state so that we explore the PV first. Please share your thoughts!

**Comment by SebLague** [comment]
Hi, the move struct is 4 bytes not 40+. It stores two ushort values, and the various properties such as start square, target square and so on are calculated from those when you access them.

**Comment by jongdetim** [comment]
Ah indeed, my bad. I must've been tired because I didn't process the const keyword for the Move masks.
